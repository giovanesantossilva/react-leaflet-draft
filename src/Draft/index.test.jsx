import { DraftControl } from '.';
import { MapContainer, Marker, FeatureGroup} from "react-leaflet";

import { cleanup, render } from "@testing-library/react";

afterEach(cleanup);

describe('MarkerMuster Component', function () {

    it('should render and unmount component', function () {
        const position = [-22.2108112, -49.6771926];
        const { container, unmount } = render(
            <MapContainer
                zoon={10}
                center={position}
            >
                <FeatureGroup>
                    <DraftControl />
                    <Marker position={position} />
                </FeatureGroup>
            </MapContainer>
        );
        expect(container).toMatchSnapshot();
        unmount();
    });

    it('should render and unmount component with draw props', function () {
        const position = [-22.2108112, -49.6771926];
        const { container, unmount } = render(
            <MapContainer
                zoon={10}
                center={position}
            >
                <FeatureGroup>
                    <DraftControl
                        draw={{
                            circle: true,
                            rectangle: true
                        }}
                    />
                    <Marker position={position} />
                </FeatureGroup>
            </MapContainer>
        );
        expect(container).toMatchSnapshot();
        unmount();
    });

    it('should render and unmount component with edit props', function () {
        const position = [-22.2108112, -49.6771926];
        const { container, unmount } = render(
            <MapContainer
                zoon={10}
                center={position}
            >
                <FeatureGroup>
                    <DraftControl
                        edit={{
                            remove: true
                        }}
                    />
                    <Marker position={position} />
                </FeatureGroup>
            </MapContainer>
        );
        expect(container).toMatchSnapshot();
        unmount();
    });

    it('should render and unmount component with position props', function () {
        const position = [-22.2108112, -49.6771926];
        const { container, unmount } = render(
            <MapContainer
                zoon={10}
                center={position}
            >
                <FeatureGroup>
                    <DraftControl
                        position='topright'
                    />
                    <Marker position={position} />
                </FeatureGroup>
            </MapContainer>
        );
        expect(container).toMatchSnapshot();
        unmount();
    });

    it('should render and unmount component with translate props', function () {
        const position = [-22.2108112, -49.6771926];
        const { container, unmount } = render(
            <MapContainer
                zoon={10}
                center={position}
            >
                <FeatureGroup>
                    <DraftControl
                        translate={{
                            toolbar: {
                                buttons: {
                                    circle: 'Circle'
                                }
                            }
                        }}
                    />
                    <Marker position={position} />
                </FeatureGroup>
            </MapContainer>
        );
        expect(container).toMatchSnapshot();
        unmount();
    });

    it('should render and unmount component with limitLayer props', function () {
        const position = [-22.2108112, -49.6771926];
        const { container, unmount } = render(
            <MapContainer
                zoon={10}
                center={position}
            >
                <FeatureGroup>
                    <DraftControl
                        limitLayers={2}
                    />
                    <Marker position={position} />
                </FeatureGroup>
            </MapContainer>
        );
        expect(container).toMatchSnapshot();
        unmount();
    });

    it('should render and unmount component with listener', function () {
        const position = [-22.2108112, -49.6771926];
        const { container, unmount } = render(
            <MapContainer
                zoon={10}
                center={position}
            >
                <FeatureGroup>
                    <DraftControl
                        onEdited={e => console.log(e)}
                        onDeleted={e => console.log(e)}
                    />
                    <Marker position={position} />
                </FeatureGroup>
            </MapContainer>
        );
        expect(container).toMatchSnapshot();
        unmount();
    });

});
