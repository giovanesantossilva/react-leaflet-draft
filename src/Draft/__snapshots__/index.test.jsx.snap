// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkerMuster Component should render and unmount component 1`] = `
<div>
  <div
    class="leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
    style="position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      />
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      >
        <div
          class="leaflet-control-zoom leaflet-bar leaflet-control"
        >
          <a
            aria-disabled="false"
            aria-label="Zoom in"
            class="leaflet-control-zoom-in"
            href="#"
            role="button"
            title="Zoom in"
          >
            <span
              aria-hidden="true"
            >
              +
            </span>
          </a>
          <a
            aria-disabled="false"
            aria-label="Zoom out"
            class="leaflet-control-zoom-out"
            href="#"
            role="button"
            title="Zoom out"
          >
            <span
              aria-hidden="true"
            >
              −
            </span>
          </a>
        </div>
        <div
          class="leaflet-draw leaflet-control"
        >
          <div
            class="leaflet-draw-section"
          >
            <div
              class="leaflet-draw-toolbar leaflet-bar leaflet-draw-toolbar-top"
            >
              <a
                class="leaflet-draw-draw-polyline"
                href="#"
                title="Draw a polyline"
              >
                <span
                  class="sr-only"
                >
                  Draw a polyline
                </span>
              </a>
              <a
                class="leaflet-draw-draw-polygon"
                href="#"
                title="Draw a polygon"
              >
                <span
                  class="sr-only"
                >
                  Draw a polygon
                </span>
              </a>
            </div>
            <ul
              class="leaflet-draw-actions"
            />
          </div>
          <div
            class="leaflet-draw-section"
          >
            <div
              class="leaflet-draw-toolbar leaflet-bar"
            >
              <a
                class="leaflet-draw-edit-edit"
                href="#"
                title="Edit layers"
              >
                <span
                  class="sr-only"
                >
                  Edit layers
                </span>
              </a>
              <a
                class="leaflet-draw-edit-remove"
                href="#"
                title="Delete layers"
              >
                <span
                  class="sr-only"
                >
                  Delete layers
                </span>
              </a>
            </div>
            <ul
              class="leaflet-draw-actions"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JavaScript library for interactive maps"
          >
            Leaflet
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MarkerMuster Component should render and unmount component with draw props 1`] = `
<div>
  <div
    class="leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
    style="position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      />
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      >
        <div
          class="leaflet-control-zoom leaflet-bar leaflet-control"
        >
          <a
            aria-disabled="false"
            aria-label="Zoom in"
            class="leaflet-control-zoom-in"
            href="#"
            role="button"
            title="Zoom in"
          >
            <span
              aria-hidden="true"
            >
              +
            </span>
          </a>
          <a
            aria-disabled="false"
            aria-label="Zoom out"
            class="leaflet-control-zoom-out"
            href="#"
            role="button"
            title="Zoom out"
          >
            <span
              aria-hidden="true"
            >
              −
            </span>
          </a>
        </div>
        <div
          class="leaflet-draw leaflet-control"
        >
          <div
            class="leaflet-draw-section"
          >
            <div
              class="leaflet-draw-toolbar leaflet-bar leaflet-draw-toolbar-top"
            >
              <a
                class="leaflet-draw-draw-polyline"
                href="#"
                title="Draw a polyline"
              >
                <span
                  class="sr-only"
                >
                  Draw a polyline
                </span>
              </a>
              <a
                class="leaflet-draw-draw-polygon"
                href="#"
                title="Draw a polygon"
              >
                <span
                  class="sr-only"
                >
                  Draw a polygon
                </span>
              </a>
              <a
                class="leaflet-draw-draw-rectangle"
                href="#"
                title="Draw a rectangle"
              >
                <span
                  class="sr-only"
                >
                  Draw a rectangle
                </span>
              </a>
              <a
                class="leaflet-draw-draw-circle"
                href="#"
                title="Draw a circle"
              >
                <span
                  class="sr-only"
                >
                  Draw a circle
                </span>
              </a>
            </div>
            <ul
              class="leaflet-draw-actions"
            />
          </div>
          <div
            class="leaflet-draw-section"
          >
            <div
              class="leaflet-draw-toolbar leaflet-bar"
            >
              <a
                class="leaflet-draw-edit-edit"
                href="#"
                title="Edit layers"
              >
                <span
                  class="sr-only"
                >
                  Edit layers
                </span>
              </a>
              <a
                class="leaflet-draw-edit-remove"
                href="#"
                title="Delete layers"
              >
                <span
                  class="sr-only"
                >
                  Delete layers
                </span>
              </a>
            </div>
            <ul
              class="leaflet-draw-actions"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JavaScript library for interactive maps"
          >
            Leaflet
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MarkerMuster Component should render and unmount component with edit props 1`] = `
<div>
  <div
    class="leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
    style="position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      />
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      >
        <div
          class="leaflet-control-zoom leaflet-bar leaflet-control"
        >
          <a
            aria-disabled="false"
            aria-label="Zoom in"
            class="leaflet-control-zoom-in"
            href="#"
            role="button"
            title="Zoom in"
          >
            <span
              aria-hidden="true"
            >
              +
            </span>
          </a>
          <a
            aria-disabled="false"
            aria-label="Zoom out"
            class="leaflet-control-zoom-out"
            href="#"
            role="button"
            title="Zoom out"
          >
            <span
              aria-hidden="true"
            >
              −
            </span>
          </a>
        </div>
        <div
          class="leaflet-draw leaflet-control"
        >
          <div
            class="leaflet-draw-section"
          >
            <div
              class="leaflet-draw-toolbar leaflet-bar leaflet-draw-toolbar-top"
            >
              <a
                class="leaflet-draw-draw-polyline"
                href="#"
                title="Draw a polyline"
              >
                <span
                  class="sr-only"
                >
                  Draw a polyline
                </span>
              </a>
              <a
                class="leaflet-draw-draw-polygon"
                href="#"
                title="Draw a polygon"
              >
                <span
                  class="sr-only"
                >
                  Draw a polygon
                </span>
              </a>
            </div>
            <ul
              class="leaflet-draw-actions"
            />
          </div>
          <div
            class="leaflet-draw-section"
          >
            <div
              class="leaflet-draw-toolbar leaflet-bar"
            >
              <a
                class="leaflet-draw-edit-edit"
                href="#"
                title="Edit layers"
              >
                <span
                  class="sr-only"
                >
                  Edit layers
                </span>
              </a>
              <a
                class="leaflet-draw-edit-remove"
                href="#"
                title="Delete layers"
              >
                <span
                  class="sr-only"
                >
                  Delete layers
                </span>
              </a>
            </div>
            <ul
              class="leaflet-draw-actions"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JavaScript library for interactive maps"
          >
            Leaflet
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MarkerMuster Component should render and unmount component with limitLayer props 1`] = `
<div>
  <div
    class="leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
    style="position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      />
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      >
        <div
          class="leaflet-control-zoom leaflet-bar leaflet-control"
        >
          <a
            aria-disabled="false"
            aria-label="Zoom in"
            class="leaflet-control-zoom-in"
            href="#"
            role="button"
            title="Zoom in"
          >
            <span
              aria-hidden="true"
            >
              +
            </span>
          </a>
          <a
            aria-disabled="false"
            aria-label="Zoom out"
            class="leaflet-control-zoom-out"
            href="#"
            role="button"
            title="Zoom out"
          >
            <span
              aria-hidden="true"
            >
              −
            </span>
          </a>
        </div>
        <div
          class="leaflet-draw leaflet-control"
        >
          <div
            class="leaflet-draw-section"
          >
            <div
              class="leaflet-draw-toolbar leaflet-bar leaflet-draw-toolbar-top"
            >
              <a
                class="leaflet-draw-draw-polyline"
                href="#"
              >
                <span
                  class="sr-only"
                />
              </a>
              <a
                class="leaflet-draw-draw-polygon"
                href="#"
              >
                <span
                  class="sr-only"
                />
              </a>
            </div>
            <ul
              class="leaflet-draw-actions"
            />
          </div>
          <div
            class="leaflet-draw-section"
          >
            <div
              class="leaflet-draw-toolbar leaflet-bar"
            >
              <a
                class="leaflet-draw-edit-edit"
                href="#"
                title="Edit layers"
              >
                <span
                  class="sr-only"
                >
                  Edit layers
                </span>
              </a>
              <a
                class="leaflet-draw-edit-remove"
                href="#"
                title="Delete layers"
              >
                <span
                  class="sr-only"
                >
                  Delete layers
                </span>
              </a>
            </div>
            <ul
              class="leaflet-draw-actions"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JavaScript library for interactive maps"
          >
            Leaflet
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MarkerMuster Component should render and unmount component with listener 1`] = `
<div>
  <div
    class="leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
    style="position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      />
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      >
        <div
          class="leaflet-control-zoom leaflet-bar leaflet-control"
        >
          <a
            aria-disabled="false"
            aria-label="Zoom in"
            class="leaflet-control-zoom-in"
            href="#"
            role="button"
            title="Zoom in"
          >
            <span
              aria-hidden="true"
            >
              +
            </span>
          </a>
          <a
            aria-disabled="false"
            aria-label="Zoom out"
            class="leaflet-control-zoom-out"
            href="#"
            role="button"
            title="Zoom out"
          >
            <span
              aria-hidden="true"
            >
              −
            </span>
          </a>
        </div>
        <div
          class="leaflet-draw leaflet-control"
        >
          <div
            class="leaflet-draw-section"
          >
            <div
              class="leaflet-draw-toolbar leaflet-bar leaflet-draw-toolbar-top"
            >
              <a
                class="leaflet-draw-draw-polyline"
                href="#"
              >
                <span
                  class="sr-only"
                />
              </a>
              <a
                class="leaflet-draw-draw-polygon"
                href="#"
              >
                <span
                  class="sr-only"
                />
              </a>
            </div>
            <ul
              class="leaflet-draw-actions"
            />
          </div>
          <div
            class="leaflet-draw-section"
          >
            <div
              class="leaflet-draw-toolbar leaflet-bar"
            >
              <a
                class="leaflet-draw-edit-edit"
                href="#"
                title="Edit layers"
              >
                <span
                  class="sr-only"
                >
                  Edit layers
                </span>
              </a>
              <a
                class="leaflet-draw-edit-remove"
                href="#"
                title="Delete layers"
              >
                <span
                  class="sr-only"
                >
                  Delete layers
                </span>
              </a>
            </div>
            <ul
              class="leaflet-draw-actions"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JavaScript library for interactive maps"
          >
            Leaflet
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MarkerMuster Component should render and unmount component with position props 1`] = `
<div>
  <div
    class="leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
    style="position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      />
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      >
        <div
          class="leaflet-control-zoom leaflet-bar leaflet-control"
        >
          <a
            aria-disabled="false"
            aria-label="Zoom in"
            class="leaflet-control-zoom-in"
            href="#"
            role="button"
            title="Zoom in"
          >
            <span
              aria-hidden="true"
            >
              +
            </span>
          </a>
          <a
            aria-disabled="false"
            aria-label="Zoom out"
            class="leaflet-control-zoom-out"
            href="#"
            role="button"
            title="Zoom out"
          >
            <span
              aria-hidden="true"
            >
              −
            </span>
          </a>
        </div>
      </div>
      <div
        class="leaflet-top leaflet-right"
      >
        <div
          class="leaflet-draw leaflet-control"
        >
          <div
            class="leaflet-draw-section"
          >
            <div
              class="leaflet-draw-toolbar leaflet-bar leaflet-draw-toolbar-top"
            >
              <a
                class="leaflet-draw-draw-polyline"
                href="#"
                title="Draw a polyline"
              >
                <span
                  class="sr-only"
                >
                  Draw a polyline
                </span>
              </a>
              <a
                class="leaflet-draw-draw-polygon"
                href="#"
                title="Draw a polygon"
              >
                <span
                  class="sr-only"
                >
                  Draw a polygon
                </span>
              </a>
            </div>
            <ul
              class="leaflet-draw-actions"
            />
          </div>
          <div
            class="leaflet-draw-section"
          >
            <div
              class="leaflet-draw-toolbar leaflet-bar"
            >
              <a
                class="leaflet-draw-edit-edit"
                href="#"
                title="Edit layers"
              >
                <span
                  class="sr-only"
                >
                  Edit layers
                </span>
              </a>
              <a
                class="leaflet-draw-edit-remove"
                href="#"
                title="Delete layers"
              >
                <span
                  class="sr-only"
                >
                  Delete layers
                </span>
              </a>
            </div>
            <ul
              class="leaflet-draw-actions"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JavaScript library for interactive maps"
          >
            Leaflet
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MarkerMuster Component should render and unmount component with translate props 1`] = `
<div>
  <div
    class="leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
    style="position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      />
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      >
        <div
          class="leaflet-control-zoom leaflet-bar leaflet-control"
        >
          <a
            aria-disabled="false"
            aria-label="Zoom in"
            class="leaflet-control-zoom-in"
            href="#"
            role="button"
            title="Zoom in"
          >
            <span
              aria-hidden="true"
            >
              +
            </span>
          </a>
          <a
            aria-disabled="false"
            aria-label="Zoom out"
            class="leaflet-control-zoom-out"
            href="#"
            role="button"
            title="Zoom out"
          >
            <span
              aria-hidden="true"
            >
              −
            </span>
          </a>
        </div>
        <div
          class="leaflet-draw leaflet-control"
        >
          <div
            class="leaflet-draw-section"
          >
            <div
              class="leaflet-draw-toolbar leaflet-bar leaflet-draw-toolbar-top"
            >
              <a
                class="leaflet-draw-draw-polyline"
                href="#"
              >
                <span
                  class="sr-only"
                />
              </a>
              <a
                class="leaflet-draw-draw-polygon"
                href="#"
              >
                <span
                  class="sr-only"
                />
              </a>
            </div>
            <ul
              class="leaflet-draw-actions"
            />
          </div>
          <div
            class="leaflet-draw-section"
          >
            <div
              class="leaflet-draw-toolbar leaflet-bar"
            >
              <a
                class="leaflet-draw-edit-edit"
                href="#"
                title="Edit layers"
              >
                <span
                  class="sr-only"
                >
                  Edit layers
                </span>
              </a>
              <a
                class="leaflet-draw-edit-remove"
                href="#"
                title="Delete layers"
              >
                <span
                  class="sr-only"
                >
                  Delete layers
                </span>
              </a>
            </div>
            <ul
              class="leaflet-draw-actions"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JavaScript library for interactive maps"
          >
            Leaflet
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
